File: CoreJava/SOLID-princples-app/src/lsp/violation/Square.java
package lsp.violation;

public class Square extends Rectangle {
	public Square(int side) {
		super(side, side);
		this.side = side;
	}

	private int side;

	@Override
	public void setWidth(int width) {
		this.width = this.height= width;
	}

	@Override
	public void setHeight(int height) {
		this.width = this.height= height;
	}
}

File: DotNET/Solid Principles/LSPViolationApp/LSPViolationApp/Square.cs
using System;

namespace LSPViolationApp
{
    class Square : Rectangle
    {
        private int _side;

        public Square(int side) : base(side, side)
        {
            _side = side;
        }

        public override int Height
        {
            set
            {
                _height = _width = value;
            }
        }

        public override int Width
        {
            set
            {
                _width = _height = value;
            }
        }
    }
}
